build/tutorials/tutorials-contest/red-dev-avalanche-chainlink-tutorial/README.md

```diff
@@ -0,0 +1,441 @@
+      }
+    }
+  ],
+  "tasks": [
+    {
+      "type": "avalanchenodejs",
+      "params": {
+        "chain": "P",
+        "method": "platform.getCurrentSupply",
+        "params": {
+        }
+      }
+    },
+    {
+      "type": "jsonparse"
+    },
+    {
+      "type": "ethuint256"
+    },
+    {
+      "type": "ethtx"
+    }
+  ]
+}
+```
+
+* jsonparse, ethuint256 & ethtx are core adaptors and are executed synchronously. For more details, you will find documentation for each adapter's usage [here](https://docs.chain.link/docs/core-adapters/).
+* runlog - By adding the address parameter, you make the event filter of the RunLog initiator more restrictive, only listening for events from that address, instead of any address. By adding the requesters parameter, you only allow requests to come from an address within the array. Please have a look at this [video](https://www.youtube.com/watch?v=ZB3GLtQvgME&38m38s) to learn how to get your oracle address.
+
+
+# Smart contracts
+
+This section will explain the smart contracts we have written for communicating with the Chainlink node through oracle address and job id.
+
+Below is the code snippet which you can copy and paste into the Remix IDE and then compile and deploy.
+
+```
+
+pragma solidity 0.4.24;
+
+import "https://github.com/smartcontractkit/chainlink/contracts/src/v0.4/ChainlinkClient.sol";
+import "https://github.com/smartcontractkit/chainlink/contracts/src/v0.4/vendor/Ownable.sol";
+
+contract ATestnetConsumer is ChainlinkClient, Ownable {
+  uint256 constant private ORACLE_PAYMENT = 1 * LINK;
+  uint256 public supply;
+  
+  event RequestAvaxSupplyFulfilled(
+    bytes32 indexed requestId,
+    uint256 indexed supply
+  );
+  
+  constructor() public Ownable() {
+    setChainlinkToken(0x0b9d5D9136855f6FEc3c0993feE6E9CE8a297846);
+  }
+  
+  function requestCurrentAvaxSupply(address _oracle, string _jobId)
+    public
+    onlyOwner
+  {
+      Chainlink.Request memory req = buildChainlinkRequest(stringToBytes32(_jobId), this, this.fulfillAvaxSupply.selector);
+      req.add("chain", "P");
+      req.add("method", "platform.getCurrentSupply");
+      req.add("path", "supply");
+      sendChainlinkRequestTo(_oracle, req, ORACLE_PAYMENT);
+  }
+  
+  function fulfillAvaxSupply(bytes32 _requestId, uint256 _supply)
+    public
+    recordChainlinkFulfillment(_requestId)
+  {
+    emit RequestAvaxSupplyFulfilled(_requestId, _supply);
+    supply = _supply;
+  }
+
+  function getChainlinkToken() public view returns (address) {
+    return chainlinkTokenAddress();
+  }
+
+  function stringToBytes32(string memory source) private pure returns (bytes32 result) {
+    bytes memory tempEmptyStringTest = bytes(source);
+    if (tempEmptyStringTest.length == 0) {
+      return 0x0;
+    }
+
+    assembly { // solhint-disable-line no-inline-assembly
+      result := mload(add(source, 32))
+    }
+  }
+
+}
+```
+Deployed smart contracts:
+
+![smartcontracts-deployed](img/smartContract-deploy.png)
```

Comment:
Please rename all the image files so they have a common and unique prefix that can identify them as belonging to this tutorial (GitBook tool we use puts all the images in a single direcory, this helps them be grouped together). For example
chainlink-tutorial-00-name-of-first-image.png,
chainlink-tutorial-01-name-of-second-image.png,
...

Reply:
Updated image names as suggested. Please have a look at 4ba5be59729a9bc2d8bf6a840c768ba00355cd4b